src/opm/input/eclipse/EclipseState/Tables/TableManager.cpp

```diff
@@ -1582,13 +1582,25 @@ std::optional<JFunc> make_jfunc(const Deck& deck) {
         for (size_t tableIdx = 0; tableIdx < tableKeyword.size(); ++tableIdx) {
             const auto& dataItem = tableKeyword.getRecord( tableIdx ).getItem("DATA");
             if (dataItem.data_size() > 0) {
-                std::shared_ptr<TableType> table = std::make_shared<TableType>( dataItem, tableIdx );
-                container.addTable( tableIdx , table );
-                lastComplete = tableIdx;
+                try {
+                    std::shared_ptr<TableType> table = std::make_shared<TableType>( dataItem, tableIdx );
+                    container.addTable( tableIdx , table );
+                    lastComplete = tableIdx;
+                } catch (const std::runtime_error& err) {
+                    throw OpmInputError(err, tableKeyword.location());
+                } catch (const std::invalid_argument& err) {
+                    throw OpmInputError(err, tableKeyword.location());
+                }
             }
             else if (tableIdx > static_cast<size_t>(0)) {
-                const auto& item = tableKeyword.getRecord(lastComplete).getItem("DATA");
-                container.addTable(tableIdx, std::make_shared<TableType>(item, tableIdx));
+                try {
+                    const auto& item = tableKeyword.getRecord(lastComplete).getItem("DATA");
+                    container.addTable(tableIdx, std::make_shared<TableType>(item, tableIdx));
+                } catch (const std::runtime_error& err) {
+                    throw OpmInputError(err, tableKeyword.location());
+                } catch (const std::invalid_argument& err) {
+                    throw OpmInputError(err, tableKeyword.location());
+                }
```

Comment:
You don't need the `try`/`catch` treatment here.  If we end up in this branch (table is empty but `tableIdx > 0`), then we're creating a copy of an already fully formed and checked table so the error would already have been raised.

Reply:
Ah, good, then I can remove that one.